Source generator doesn't work in Unity

#### Bug description

Source generator doesn't work in Unity because it references System.Collections.Immutable 8.0.0. After changing it to reference System.Collections.Immutable 6.0.0 locally, it works correctly. 

#### Additional context
Environment:
* MessagePack: 3.0.300
* Unity: 2022.3.40

The following code can reproduce the issue in a new Unity project:
```cs
using MessagePack;
using UnityEngine;

[MessagePackObject]
public class A
{
    [Key(0)]
    public int X;
}

public class Test : MonoBehaviour
{
    public void Start()
    {
        Debug.Log(GeneratedMessagePackResolver.Instance);
    }
}
```

Error:
```
Assets\Test.cs(17,19): error CS0103: The name 'GeneratedMessagePackResolver' does not exist in the current context
```

The code works when using MessagePack 3.0.238-rc.1.

It seems to be a regression as it was resolved before: https://github.com/MessagePack-CSharp/MessagePack-CSharp/pull/1734#issuecomment-2182213787

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -61,7 +61,7 @@
   <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.1'">
     <PackageVersion Include="System.Collections.Immutable" Version="8.0.0" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' or '$(TargetFramework)' == 'net472'">
+  <ItemGroup Condition=" ('$(TargetFramework)' == 'netstandard2.0' and '$(IsAnalyzerProject)' != 'true') or '$(TargetFramework)' == 'net472'">
     <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0" />
     <PackageVersion Include="System.Collections.Immutable" Version="8.0.0" />
     <PackageVersion Include="System.Memory" Version="4.5.5" />

